DAQ_System/Source/Adapter/DataLogger/SdDataLogger.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+* SD Data Logger
+* Authors:        Cristian Cruz
+*                 Ethan Fischer
+*
+* Email:          cris14.alex@gmail.com
+*                 ejf180001@utdallas.edu
+* 
+* (c) 2022 Dallas Formula Racing - Embedded Firmware Team
+* Formula SAE International Collegiate Chapter
+* GPL-3.0 License
+*/
+
+#ifndef DATA_LOGGER_H
+#define DATA_LOGGER_H
+
+#include "mbed.h"
+#include "FATFileSystem.h"
+#include "SDBlockDevice.h"
+
+#include "Application/I_Data_Logger.hpp"
+
+namespace adapter {
+
+class SdDataLogger : public application::I_Data_Logger {
+    public:
+        SdDataLogger(PinName mosi, PinName miso, PinName sck, PinName cs);
+        virtual ~SdDataLogger();
+        
+        virtual uint8_t Mount(FileSystem*, BlockDevice*) override;
+        virtual uint8_t Unmount(FileSystem*) override;
+        virtual uint8_t FileOpen(FILE**, char*) override;
+        virtual uint8_t FileClose(FILE*) override;
+        virtual uint8_t FileWrite(FILE*, const char*) override;
+
+        char write_buffer; // write buffer
+    
+    private:
+        SDBlockDevice block_device; // physical block device
+        FATFileSystem file_system; // file system
+        int log_session;
+        char file_name[16];
+        const int kBlockSectorByteSize;
```

Comment:
Append an underscore in the name of the class Do this for the `write_buffer`, too. This is to differentiate between a class-owned variable and other variables (this helps eliminate the use of the `this->` pointer)

Reply:
I'm not really sure what you mean here. You said "in the name of the class" but that doesn't really make sense. I assume you mean at the end of the variable names, so that's what I did, but this doesn't seem to make any difference, since we don't ever use the this-> pointer.

Reply:
Oh, that was a typo — I'm lucky you understood me correctly. Usually the `this->` pointer is used if we have variables with exact names, and we need to use the variable owned by the class. Would also like to follow the style guide consistently